### PR TITLE
neovim/init: set basic options via `vim.options`

### PR DESCRIPTION
**Pull request by NotAShelf**
Less DAGs, more customizability.

Moves interpolated basic vim options into `vim.options` with a backwards compatible interface. I'm fully intending to deprecate all of the options before merge.

## Sanity Checking

[editorconfig]: https://editorconfig.org
[changelog]: https://github.com/NotAShelf/nvf/tree/main/docs/release-notes

- [ ] I have updated the [changelog] as per my changes.
- [x] I have tested, and self-reviewed my code.
- Style and consistency
  - [x] I ran **Alejandra** to format my code (`nix fmt`).
  - [x] My code conforms to the [editorconfig] configuration of the project.
  - [x] My changes are consistent with the rest of the codebase.
- If new changes are particularly complex:
  - [x] My code includes comments in particularly complex areas
  - [ ] I have added a section in the manual.
  - [ ] _(For breaking changes)_ I have included a migration guide.
- Package(s) built:
  - [x] `.#nix` (default package)
  - [ ] `.#maximal`
  - [x] `.#docs-html`
- Tested on platform(s)
  - [x] `x86_64-linux`
  - [ ] `aarch64-linux`
  - [ ] `x86_64-darwin`
  - [ ] `aarch64-darwin`

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: modules/extra/deprecations.nix [hunk 1]
```diff
@@ -1,65 +1,95 @@
...
+    updateTime = "updatetime";
+    mapTimeout = "tm";
+    cursorlineOpt = "cursorlineopt";
+    splitBelow = "splitbelow";
+    splitRight = "splitright";
+    autoIndent = "autoindent";
+    wordWrap = "wrap";
+  };
+
+  renamedVimOpts = batchRenameOptions ["vim"] ["vim" "options"] migrationTable;
```
Comment (diniamo) [inline modules/extra/deprecations.nix:19]: I'd rather just inline migrationTable

**Review by diniamo — CHANGES_REQUESTED** [review]
Missing a breaking changes entry in the release notes.

I'm also in favor of removing the more complex options.

**Comment by NotAShelf** [comment]
It is not *entirely* a breaking change since the side effects have been remedied. All renamed options have an equivalent in the new format, except for shift/tab width.

I'll add an entry for those.

**Comment by diniamo** [comment]
This definitely *is* a breaking changes, since the renamed options will break the user's configuration if they used to override any of them.

**Comment by NotAShelf** [comment]
Moved the notes to the breaking changes section.

**Review by diniamo — APPROVED** [review]
Looking good

**Pull request merged**